core/ir/aarch64/codec.c

```diff
@@ -1484,6 +1511,55 @@ encode_opnd_imm4(uint enc, int opcode, byte *pc, opnd_t opnd, OUT uint *enc_out)
     return encode_opnd_int(8, 4, false, 0, 0, opnd, enc_out);
 }
 
+/* cmode4_s_sz_msl: Operand for 32 bit elements' shift amount (shifting ones) */
+
+static inline bool
+decode_opnd_cmode4_s_sz_msl(uint enc, int opcode, byte *pc, OUT opnd_t *opnd)
+{
+    /* cmode size amounts
+     * 1110  8    0
+     * 10x0  16   0,8
+     * 0xx0  32   0,8,16,24
+     * 110x  32   8,16 (MSL, shifting ones)
+     */
+    int cmode4 = extract_uint(enc, 12, 1);
+    int esize = 32;
+    int shift = ((cmode4 == 0) ? 8 : 16) | (1 << 28);
```

Comment:
> I'm not following how this matches the table: so this is the bottom 2 rows, or just the bottom row? Maybe indicate that in the comment.

Just the bottom row, 32 but `MSL` rather than `LSL`. Removed other entries and added explanation for `MSL`. The `MSL` operator stands for 'Modified Shift Left', a left shift which shifts ones instead of zeros into the low order bits, unlike the more common/default `LSL` (Logical Shift Left) which shifts in zeros.

> Where does the 1<<28 come from? Shouldn't this just be the 8 or 16 if it's the bottom row?

This is the element size, 32 bits, because size and shift amount encapsulated in one int operand, see my reply below.